test/app/components/sub_component_component.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class SubComponentComponent < ViewComponent::Base
+  include ViewComponent::SubComponents
+
+  renders_one :title
+  renders_one :subtitle
+  renders_one :footer, -> (classes: "", &block) do
+    content_tag :footer, class: "footer #{classes}" do
+      block.call if block
```

Comment:
Perhaps we could come up with a helper/abstraction to avoid this `block.call if block` situation? This is a leak in the API to the developer.